kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveKeyValueBytesStoreSupplier.java

```diff
@@ -26,18 +27,28 @@ public class ResponsiveKeyValueBytesStoreSupplier implements KeyValueBytesStoreS
 
   private final ResponsiveKeyValueParams params;
 
+  private boolean isTimestamped;
```

Comment:
@agavra I know this is super awkward but I wanted to be as non-intrusive as possible, and most importantly, not break things for Metronome.

Of course they'll have to make a few (very minor) code changes to convert to an async processor anyways, so a few more wouldn't hurt. That said, I'd still rather keep the current API where users only need to specify whether it's timestamped or not via the builder, and don't need to worry about having the store supplier match the builder in terms of using the correct API in ResponsiveStores. I don't feel super strongly one way or another though so I'm happy to delegate if you want to design the API differently

Reply:
I've always been a fan of copying Kafka Streams, even if it isn't ideal so they can just `s/Stores/ResponsiveStores/g` - i.e. it even took me a second to migrate "word count" from the Kafka Streams example because I had to figure out the ResponsiveStores API. It's not a big deal, but I think the more we lean on "Responsive == Kafka Streams" the smoother the transition feels and the easier it is to write/maintain migration docs.